x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -166,12 +166,23 @@ func createCloudWatchEvents(getMetricDataResults []cloudwatch.MetricDataResult,
 
 	// AWS EC2 Metrics
 	mapOfMetricSetFieldResults := make(map[string]interface{})
-	for _, output := range getMetricDataResults {
-		if len(output.Values) == 0 {
-			continue
+
+	// Find a timestamp for all metrics in output
+	timestamp := aws.FindTimestamp(getMetricDataResults)
+	if !timestamp.IsZero() {
```

Comment:
Looks like it it's still possible that we have zero timestamp? This is when all events miss the timestamp field?

Reply:
Yes, when there is no data returned from Cloudwatch basically.